rpcs3/Emu/NP/rpcn_client.cpp

```diff
@@ -2192,19 +2192,298 @@ namespace rpcn
+		bool privilege_grant = false;
+		bool stealth = false;
+
+		std::vector<flatbuffers::Offset<MatchingAttr>> vec_attrs;
+
+		for (const SceNpMatchingAttr* cur_attr = attr_list; cur_attr != nullptr; cur_attr = cur_attr->next ? cur_attr->next.get_ptr() : nullptr)
+		{
+			ensure(cur_attr->type == SCE_NP_MATCHING_ATTR_TYPE_BASIC_NUM || cur_attr->type == SCE_NP_MATCHING_ATTR_TYPE_GAME_BIN ||
+					   cur_attr->type == SCE_NP_MATCHING_ATTR_TYPE_GAME_NUM,
+				"Invalid attribute type in createjoin_room_gui");
+
+			switch (cur_attr->type)
+			{
+			case SCE_NP_MATCHING_ATTR_TYPE_BASIC_NUM:
+			{
+				ensure(cur_attr->id == SCE_NP_MATCHING_ROOM_ATTR_ID_TOTAL_SLOT || cur_attr->id == SCE_NP_MATCHING_ROOM_ATTR_ID_PRIVATE_SLOT ||
+						   cur_attr->id == SCE_NP_MATCHING_ROOM_ATTR_ID_PRIVILEGE_TYPE || cur_attr->id == SCE_NP_MATCHING_ROOM_ATTR_ID_ROOM_SEARCH_FLAG,
+					"Invalid basic num attribute id");
+
+				switch (cur_attr->id)
+				{
+				case SCE_NP_MATCHING_ROOM_ATTR_ID_TOTAL_SLOT:
+					total_slots = cur_attr->value.num;
+					break;
+				case SCE_NP_MATCHING_ROOM_ATTR_ID_PRIVATE_SLOT:
+					private_slots = cur_attr->value.num;
+					break;
+				case SCE_NP_MATCHING_ROOM_ATTR_ID_PRIVILEGE_TYPE:
+					ensure(cur_attr->value.num == SCE_NP_MATCHING_ROOM_PRIVILEGE_TYPE_NO_AUTO_GRANT || cur_attr->value.num == SCE_NP_MATCHING_ROOM_PRIVILEGE_TYPE_AUTO_GRANT, "Invalid SCE_NP_MATCHING_ROOM_ATTR_ID_PRIVILEGE_TYPE value");
+					privilege_grant = (cur_attr->value.num == SCE_NP_MATCHING_ROOM_PRIVILEGE_TYPE_AUTO_GRANT);
+					break;
+				case SCE_NP_MATCHING_ROOM_ATTR_ID_ROOM_SEARCH_FLAG:
+					ensure(cur_attr->value.num == SCE_NP_MATCHING_ROOM_SEARCH_FLAG_OPEN || cur_attr->value.num == SCE_NP_MATCHING_ROOM_SEARCH_FLAG_STEALTH, "Invalid SCE_NP_MATCHING_ROOM_ATTR_ID_ROOM_SEARCH_FLAG value");
+					stealth = (cur_attr->value.num == SCE_NP_MATCHING_ROOM_SEARCH_FLAG_STEALTH);
+					break;
+				default:
+					fmt::throw_exception("Unreachable");
+					break;
+				}
+
+				break;
+			}
+			case SCE_NP_MATCHING_ATTR_TYPE_GAME_BIN:
+			{
+				ensure(cur_attr->id >= 1u && cur_attr->id <= 16u, "Invalid game bin attribute id");
+				ensure(cur_attr->value.data.size <= 64u || ((cur_attr->id == 1u || cur_attr->id == 2u) && cur_attr->value.data.size <= 256u), "Invalid game bin size");
+
+				const std::vector<u8> vec_data(static_cast<const u8*>(cur_attr->value.data.ptr.get_ptr()), static_cast<const u8*>(cur_attr->value.data.ptr.get_ptr()) + cur_attr->value.data.size);
+				auto attr = CreateMatchingAttrDirect(builder, cur_attr->type, cur_attr->id, 0, &vec_data);
+				vec_attrs.push_back(attr);
+				break;
+			}
+			case SCE_NP_MATCHING_ATTR_TYPE_GAME_NUM:
+			{
+				ensure(cur_attr->id >= 1u && cur_attr->id <= 16u, "Invalid game num attribute id");
+
+				auto attr = CreateMatchingAttrDirect(builder, cur_attr->type, cur_attr->id, cur_attr->value.num, nullptr);
+				vec_attrs.push_back(attr);
+				break;
+			}
+			default:
+				fmt::throw_exception("Unreachable");
```

Comment:
Remove ensure above and do
```suggestion
				fmt::throw_exception("Invalid attribute type in createjoin_room_gui");
```